icebox/icebox.py

```diff
@@ -1619,6 +1671,99 @@ def parse_db(text, device="1k"):
+        # 3'b011 = "SB_PLL40_CORE"
+        # 3'b111 = "SB_PLL40_2F_CORE"
+        "PLLTYPE_0":            (12,  0, "PLLCONFIG_5"),
+        "PLLTYPE_1":            (14,  0, "PLLCONFIG_1"),
+        "PLLTYPE_2":            (14,  0, "PLLCONFIG_3"),
+
+        # 3'b000 = "DELAY"
+        # 3'b001 = "SIMPLE"
+        # 3'b010 = "PHASE_AND_DELAY"
+        # 3'b110 = "EXTERNAL"
+        "FEEDBACK_PATH_0":      (14,  0, "PLLCONFIG_5"),
+        "FEEDBACK_PATH_1":      (11,  0, "PLLCONFIG_9"),
+        "FEEDBACK_PATH_2":      (12,  0, "PLLCONFIG_1"),
+
+        # 1'b0 = "FIXED"
+        # 1'b1 = "DYNAMIC" (also set FDA_FEEDBACK=4'b1111)
+        "DELAY_ADJMODE_FB":     (13,  0, "PLLCONFIG_4"),
+
+        # 1'b0 = "FIXED"
+        # 1'b1 = "DYNAMIC" (also set FDA_RELATIVE=4'b1111)
+        "DELAY_ADJMODE_REL":    (13,  0, "PLLCONFIG_9"),
+
+        # 2'b00 = "GENCLK"
+        # 2'b01 = "GENCLK_HALF"
+        # 2'b10 = "SHIFTREG_90deg"
+        # 2'b11 = "SHIFTREG_0deg"
+        "PLLOUT_SELECT_A_0":    (12,  0, "PLLCONFIG_6"),
+        "PLLOUT_SELECT_A_1":    (12,  0, "PLLCONFIG_7"),
+        # 2'b00 = "GENCLK"
+        # 2'b01 = "GENCLK_HALF"
+        # 2'b10 = "SHIFTREG_90deg"
+        # 2'b11 = "SHIFTREG_0deg"
+        "PLLOUT_SELECT_B_0":    (12,  0, "PLLCONFIG_2"),
+        "PLLOUT_SELECT_B_1":    (12,  0, "PLLCONFIG_3"),
+
+        # Numeric Parameters
+        "SHIFTREG_DIV_MODE":    (12,  0, "PLLCONFIG_4"),
+        "FDA_FEEDBACK_0":       (12,  0, "PLLCONFIG_9"),
+        "FDA_FEEDBACK_1":       (13,  0, "PLLCONFIG_1"),
+        "FDA_FEEDBACK_2":       (13,  0, "PLLCONFIG_2"),
+        "FDA_FEEDBACK_3":       (13,  0, "PLLCONFIG_3"),
+        "FDA_RELATIVE_0":       (13,  0, "PLLCONFIG_5"),
+        "FDA_RELATIVE_1":       (13,  0, "PLLCONFIG_6"),
+        "FDA_RELATIVE_2":       (13,  0, "PLLCONFIG_7"),
+        "FDA_RELATIVE_3":       (13,  0, "PLLCONFIG_8"),
+        "DIVR_0":               (10,  0, "PLLCONFIG_1"),
+        "DIVR_1":               (10,  0, "PLLCONFIG_2"),
+        "DIVR_2":               (10,  0, "PLLCONFIG_3"),
+        "DIVR_3":               (10,  0, "PLLCONFIG_4"),
+        "DIVF_0":               (10,  0, "PLLCONFIG_5"),
+        "DIVF_1":               (10,  0, "PLLCONFIG_6"),
+        "DIVF_2":               (10,  0, "PLLCONFIG_7"),
+        "DIVF_3":               (10,  0, "PLLCONFIG_8"),
+        "DIVF_4":               (10,  0, "PLLCONFIG_9"),
+        "DIVF_5":               (11,  0, "PLLCONFIG_1"),
+        "DIVF_6":               (11,  0, "PLLCONFIG_2"),
+        "DIVQ_0":               (11,  0, "PLLCONFIG_3"),
+        "DIVQ_1":               (11,  0, "PLLCONFIG_4"),
+        "DIVQ_2":               (11,  0, "PLLCONFIG_5"),
+        "FILTER_RANGE_0":       (11,  0, "PLLCONFIG_6"),
+        "FILTER_RANGE_1":       (11,  0, "PLLCONFIG_7"),
+        "FILTER_RANGE_2":       (11,  0, "PLLCONFIG_8"),
+        "TEST_MODE":            (12,  0, "PLLCONFIG_8"),
+
+        # PLL Ports
+        # TODO(awygle) confirm these
```

Comment:
If you haven't already, the way to check these is to create a design using icecube (with pins constrained using a PCF) then run it through icebox_vlog (pass the PCF to this too to get nice net names) and look at where the routing for each signal goes in the output file.